### DISCUSSION TITLE
Import Books for existing m4b files?

### DISCUSSION THREAD

**Discussion by brabin99** [openaudible/openaudible#1478] [Q&A] [answered]
I feel like I must be missing something, but couldn't find this addressed. I am importing existing m4b files from my hard drive (that were not previously Audible books). The Import Books dialogue box doesn't have the option to show m4b files and they are grayed out. I can successfully import them using drag and drop, but would prefer to use the Import Books command.

**Comment by openaudible** [comment]
Oh, that might have been a bug that was fixed. You can also drag and drop files and directories onto OpenAudible.

**Comment by Cfomodz** [comment]
File dialog filter is hardcoded to only accept `.aax` files in `AudibleGUI.java`

```java
String ext = "*.aax";
String name = "Audible Files";
```

**Reply by Cfomodz** [reply]
**To:**
```java
String ext = "*.aax;*.m4b";
String name = "Audible Files (AAX, M4B)";
```

**Reply by openaudible** [reply]
Are you using the latest version or old open source version? 
Just drag and drop files onto the latest OpenAudible application.

**Reply by Cfomodz** [reply]
I'm talking about the dialog upload, not drag and drop. I am explaining the fix, not expressing my own problem.

**Reply by Cfomodz** [reply]
- Drag and drop already works because it bypasses the FileDialog filter
- The import pipeline (`importAAX()` method) handles the files correctly
- This fix simply allows the dialog to display M4B files for selection

**Comment by openaudible** [comment, marked as answer]
OK, this issue has been fixed in the latest binary release. 
https://openaudible.org/download
